src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1523,7 +1527,25 @@ MasterAggregateExpression(Aggref *originalAggregate,
 		columnTargetEntry = makeTargetEntry((Expr *) column, argumentId, NULL, false);
 		newMasterAggregate->args = list_make1(columnTargetEntry);
 
-		newMasterExpression = (Expr *) newMasterAggregate;
+		/* cast numeric sum result to bigint (like count) */
+		coerceExpr = makeNode(CoerceViaIO);
+		coerceExpr->arg = (Expr *) newMasterAggregate;
+		coerceExpr->resulttype = INT8OID;
+		coerceExpr->resultcollid = InvalidOid;
+		coerceExpr->coerceformat = COERCE_EXPLICIT_CALL;
```

Comment:
We could use IMPLICIT here, to avoid changing the EXPLAIN output, but that'd probably be a bit confusing.